packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -634,6 +631,12 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
     if (_lastAxisDirection == null || !_needPaint(_lastMetrics)) {
       return;
     }
+    assert(
+      textDirection != null,
+      'A TextDirection must be provided before a Scrollbar can be painted.',
+    );
+    _resolvedPadding = padding.resolve(textDirection);
```

Comment:
With this assert, it looks like `textDirection` is now effectively required?

Would it be possible to only resolve this if necessary, like if it's an EdgeInsetsGeometry and textDirection is non-null?